### PR TITLE
Add render test suite

### PR DESCRIPTION
**Pull request by chris-laplante**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -30,3 +30,8 @@ tokio = { version = "1", features = ["time", "rt"] }
 default = ["unicode-width", "console/unicode-width"]
 improved_unicode = ["unicode-segmentation", "unicode-width", "console/unicode-width"]
 in_memory = ["vt100"]
+
+[[test]]
+name = "render"
+path = "tests/render.rs"
+required-features = ["in_memory"]
```
Comment (djc) [inline Cargo.toml:37]: Instead of this, I'd prefer putting a `#![cfg(feature = "in_memory")]` at the top of `render.rs`.

**Comment by djc** [comment]
Yeah... I think we should rip out the draw limiting in the `BarState` and instead rely on the `LeakyBucket` stuff in the draw target.

**Comment by chris-laplante** [comment]
> Yeah... I think we should rip out the draw limiting in the `BarState` and instead rely on the `LeakyBucket` stuff in the draw target.

Feel free to repurpose this PR if you want to add commits to do that. It might make sense given to develop those fixes in tandem with the render test suite. I can help flesh out the test suite (as originally intended) but I'm not sure I can help much with `LeakyBucket` though.

**Comment by djc** [comment]
Do you want to test? https://github.com/console-rs/indicatif/pull/380

**Comment by djc** [comment]
I added back the test case in 3930cfc324439bb9dc8977f5380e263827b79544 after I reverted some of my draw limiting changes and merged the move into an integration test as 0c153b8bfadb28fd45ec76b284609d66e7ae8e24. Thanks!

**Pull request closed without merging**